collections/_posts/2023-07-06-Mean_teachers_are_better_role_models.md

```diff
@@ -0,0 +1,203 @@
+tags: machine-learning, neural and evolutionary computing
+author: Tarvainen Antti, Valpola Harri
+cite:
+    authors: "Tarvainen et Valpola"
+    title:   "Mean teachers are better role models: Weight-averaged consistency targets improve semi-supervised deep learning results"
+    venue:   "Neural Information Processing Systems 30 (NIPS 2017) pre-proceedings"
+pdf: https://arxiv.org/pdf/1703.01780.pdf
+---
+
+# Highlights
+
+- Method : The mean teachers model **averages model weigths**, instead of label predictions like the Temporal Ensembling method. Allows for more accurate targets and faster feedback between the student and teacher models.
+- Performances : demonstrates **enhanced performance on various datasets** (SVHN, CIFAR-10, and ImageNet) when combined with residual networks. Achieves lower error rates and higher accuracy compared to Temporal Ensembling, even when trained with fewer labeled examples.
+- Scalability and applicability : Enables working on **larger dataset**, and training with **fewer labeled examples**.
+
+# Introduction
+Limitations of deep learning models :
+- requires large number of parameters and is prone to overfitting. 
+- manual high quality labels for training data is expensive and time consuming.
+
+Interest in semi-supervised learning to use the unlabeled data effectively. 
+
+&rarr; focus on **regularization methods** in **semi-supervised learning**
+
+# Method 
+
+## Apllying noise and Ensembling method
+
+### **Noise regularization** : 
+- Add noise to the input or intermediate representations of a model
+- Helps learning abstract invariances
+- **Pushes decision boundaries away from labeled data points**  
+
+Limitations of noise regularization :
+- classification cost is **undefined for unlabeled examples** &rarr; not helpful in semi-supervised learning.
+
+&rarr; The $\Gamma$ model to overcome this
+
+### **The $\Gamma$ model [1]**
+- Evaluates each data point with and without noise, and then applies a consistency cost between the two predictions. 
+- Student teacher models where the **teacher generates targets**, which are then **used by itself as a student for learning**
+
+Limitations of $\Gamma$ model :
+- the model itself generates targets, which may be incorrect
+- the cost of inconsistency outweighs that of misclassification, preventing the learning of new information
+- can lead to a **confirmation bias**
+
+&rarr; Confirmation biais can be mitigated by improving the targets. One way is illustrated with the $\Pi$ model, selecting a different teacher model than the student model
+
+> **NOTE :** The authors mention another way : "to choose the perturbation
+of the representations carefully instead of barely applying additive or multiplicative noise" wich is the subject of another paper of [Miyato et al. 2017](https://doi.org/10.48550/arXiv.1704.03976)[2] that will be refered as VAT+EntMin latter. 
```

Comment:
Fix footnote formatting (the link to the paper should go in the footnote at the bottom):
```suggestion
of the representations carefully instead of barely applying additive or multiplicative noise" wich is the subject of another paper by Miyato _et al._ (2017)[^2] that will be refered as VAT+EntMin latter. 
```